app/api/discount_codes.py

```diff
@@ -115,9 +154,48 @@ class DiscountCodeDetail(ResourceDetail):
     Discount Code detail by id or code.
     """
 
+    def decide_schema(self, json_data):
+        """To decide discount code schema based on posted data.
+        :param json_data:
+        :return:"""
+
+        used_for = json_data['data']['attributes'].get('used-for')
+        if used_for == 'ticket':
+            self.schema = DiscountCodeSchemaTicket
+        elif used_for == 'event':
+            self.schema = DiscountCodeSchemaEvent
+
     def before_get(self, args, kwargs):
+        if kwargs.get('ticket_id'):
+            if has_access('is_coorganizer'):
+                ticket = safe_query(db, Ticket, 'id', kwargs['ticket_id'], 'ticket_id')
+                if ticket.discount_code_id:
+                    kwargs['id'] = ticket.discount_code_id
+                else:
+                    kwargs['id'] = None
+            else:
+                raise UnprocessableEntity(
+                    {'source': ''},
+                    "Please verify your permission. You must have coorganizer\
+                    privileges to view ticket discount code details")
+        if kwargs.get('event_id'):
+            if has_access('is_admin'):
+                event = safe_query(
+                    db, Event, 'id', kwargs['event_id'], 'event_id')
```

Comment:
On line 171, you have everything on one line, and here you are using two lines. Why this inconsistency. Please make it on one line.